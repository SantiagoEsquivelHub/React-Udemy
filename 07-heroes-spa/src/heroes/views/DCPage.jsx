import React from 'react'
import { HeroList } from '../components'

export const DCPage = () => {
  return (
    <div>
      <HeroList publisher='DC Comics' />
    </div>
  )
}
