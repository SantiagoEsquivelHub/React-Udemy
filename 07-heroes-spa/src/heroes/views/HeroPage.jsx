import { useParams, Navigate } from 'react-router-dom'
import { getHeroeById } from '../helpers';

export const HeroPage = () => {

  const { id } = useParams();

  const heroe = getHeroeById(id);

  if (!heroe) {
    return <Navigate to="/marvel"/>
  }

  return (
    <div>
      <h1>HeroPage</h1>
      <hr />
    </div>
  )
}
