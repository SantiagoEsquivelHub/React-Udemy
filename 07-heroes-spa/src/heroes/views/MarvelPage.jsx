import React from 'react'
import { HeroeList } from '../components'

export const MarvelPage = () => {
    return (
        <div>
            <HeroeList publisher='Marvel Comics' />
        </div>
    )
}
