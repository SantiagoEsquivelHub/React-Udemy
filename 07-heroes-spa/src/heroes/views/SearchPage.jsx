import { useLocation, useNavigate } from 'react-router-dom';
import { useForm } from '../../hooks/useForm';
import { HeroCard } from '../components'
import queryString from 'query-string';
import { getHeroesByName } from '../helpers';

export const SearchPage = () => {

  const navigate = useNavigate();
  const location = useLocation();
  const { q = '' } = queryString.parse(location.search);

  const heroes = getHeroesByName(q);

  const showSearch = (q.length === 0);
  const showError = (q.length > 0) && (heroes.length === 0);

  const { formState, onInputChange, handleReset, searchText } = useForm({
    searchText: q
  });

  const onSearchSubmit = (e) => {
    e.preventDefault();

    if (searchText.trim().length <= 1) return;

    navigate(`?q=${searchText}`);
    console.log({ q })
  }

  return (
    <div>

      <h4>Search</h4>
      <hr />

      <div className="row">
        <div className="col-5">
          <h4>Searching...</h4>
          <hr />
          <form onSubmit={onSearchSubmit} aria-label="form">
            <input
              type="text"
              placeholder='Search a heroe'
              className='form-control mb-2'
              name='searchText'
              autoComplete='off'
              value={searchText}
              onChange={onInputChange}
            />

            <button className='btn btn-outline-primary'>
              Search
            </button>
          </form>
        </div>

        <div className="col-7">
          <h4>Results</h4>
          <hr />

          <div className="alert alert-primary animate__fadeIn animate__animated" style={{ display: showSearch ? '' : 'none' }} aria-label="search_heroe">
            Search a heroe
          </div>

          <div className="alert alert-danger animate__fadeIn animate__animated" style={{ display: showError ? '' : 'none' }} aria-label="error_search_heroe">
            There's no heroe with <b>{q}</b>
          </div>


          {
            heroes.map(heroe => (
              <HeroCard key={heroe.id} {...heroe} />
            ))
          }

        </div>

      </div>
    </div >
  )
}
