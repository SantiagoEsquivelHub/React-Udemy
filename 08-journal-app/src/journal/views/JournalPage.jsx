import { Typography } from '@mui/material'
import React from 'react'
import { JournalLayout } from '../layout/JournalLayout'

export const JournalPage = () => {
  return (
    <JournalLayout>
      <Typography>dfasdfasdfasdsdfasdfasdfasdfasdfasdfasdfasdfasddddddddddddddddddddddfdfasdfasdfasdsdfasdfasdfasdfasdfasdfasdfasdfasddddddddddddddddddddddfdfasdfasdfasdsdfasdfasdfasdfasdfasdfasdfasdfasddddddddddddddddddddddfdfasdfasdfasdsdfasdfasdfasdfasdfasdfasdfasdfasddddddddddddddddddddddf</Typography>

      {/* NothingSelected */} 

      {/* NoteView */} 

    </JournalLayout>
  )
}
