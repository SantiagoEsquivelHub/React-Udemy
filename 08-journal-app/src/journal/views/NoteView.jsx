import { useEffect, useMemo } from "react"
import { Button, Grid, TextField, Typography } from "@mui/material"
import { useDispatch, useSelector } from "react-redux"
import Swal from "sweetalert2"
import moment from "moment/moment"
import { SaveOutlined } from "@mui/icons-material"
import { useForm } from "../../hooks/useForm"
import { setActiveNote } from "../../store/journal/journalSlice"
import { startSavingNotes } from "../../store/journal/thunks"
import { ImageGallery } from "../components"
import "sweetalert2/dist/sweetalert2.css";

export const NoteView = () => {

    const dispatch = useDispatch();

    const { active: note, savedMessage, isSaving } = useSelector(state => state.journal)
    const { title, body, date, onInputChange, formState } = useForm(note);

    const dateString = useMemo(() => {
        const newDate = new Date(date);
        return newDate.toUTCString();
    }, [date])

    useEffect(() => {
        dispatch(setActiveNote(formState));
    }, [formState])

    useEffect(() => {
        if (savedMessage.length > 0) {
            Swal.fire('Nota actualizada!', savedMessage, 'success')
        }
    }, [savedMessage])

    const onSaveNote = () => {
        dispatch(startSavingNotes());
    }

    return (
        <Grid
            className='animate__animated animate__fadeIn animate__faster'
            container
            direction='row'
            justifyContent='space-between'
            sx={{ mb: 1 }}
        >
            <Grid item>
                <Typography sx={{ fontSize: 39, fontWeight: "light" }}>{moment(dateString).format('LLLL')}</Typography>
            </Grid>

            <Grid item>
                <Button
                    color='primary'
                    sx={{ padding: 2 }}
                    onClick={onSaveNote}
                    disabled={isSaving}
                >
                    <SaveOutlined sx={{ fontSize: 30, mr: 1 }} />
                    Guardar
                </Button>
            </Grid>

            <Grid container>
                <TextField
                    type='text'
                    variant='filled'
                    fullWidth
                    placeholder="Ingrese un titulo"
                    label='Título'
                    sx={{ border: 'none', mb: 1 }}
                    name='title'
                    value={title}
                    onChange={onInputChange}
                />

                <TextField
                    type='text'
                    variant='filled'
                    fullWidth
                    multiline
                    placeholder="¿Qué sucedió en el día de hoy?"
                    minRows={5}
                    name='body'
                    value={body}
                    onChange={onInputChange}
                />

                <ImageGallery />


            </Grid>

        </Grid>
    )
}
