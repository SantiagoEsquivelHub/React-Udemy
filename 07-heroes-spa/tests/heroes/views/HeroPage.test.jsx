import { fireEvent, render, screen } from "@testing-library/react"
import { MemoryRouter, Route, Routes } from "react-router-dom"
import { HeroPage } from "../../../src/heroes/views/HeroPage"
import { MarvelPage } from "../../../src/heroes/views/MarvelPage";

const mockedUseNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useNavigate: () => mockedUseNavigate
}))

describe('Pruebas en <HeroPage />', () => {

    test('debe de mostrar la imagen de marvel-spider', () => {

        render(
            <MemoryRouter initialEntries={['/hero/marvel-spider']}>
                <Routes>
                    <Route path='hero/:id' element={<HeroPage />} />
                </Routes>
            </MemoryRouter>
        )

        //screen.debug();
        const imgHero = screen.getByRole('img');
        expect(imgHero.src).toContain('/assets/heroes/marvel-spider.jpg');

    })

    test('debe de navegar a la pagina anterior si se le hace click al boton de regresar', () => {

        render(
            <MemoryRouter initialEntries={['/hero/marvel-spider']}>
                <Routes>
                    <Route path='hero/:id' element={<HeroPage />} />
                </Routes>
            </MemoryRouter>
        )

        //screen.debug();
        const btnBack = screen.getByRole('button');
        fireEvent.click(btnBack);
        expect(mockedUseNavigate).toHaveBeenCalledWith(-1);
    })

    test('debe de navegar a la pagina de /marvel si se detecta el id de un heroe no existente', () => {

        const { container } = render(
            <MemoryRouter initialEntries={['/hero/marvel-spider2']}>
                <Routes>
                    <Route path='hero/:id' element={<HeroPage />} />
                    <Route path='marvel' element={<MarvelPage />} />
                </Routes>
            </MemoryRouter>
        )

        //screen.debug();
        expect(container).toMatchSnapshot();
    })

})