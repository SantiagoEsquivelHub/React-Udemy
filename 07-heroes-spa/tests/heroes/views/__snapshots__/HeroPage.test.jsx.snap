// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HeroPage /> debe de navegar a la pagina de /marvel si se detecta el id de un heroe no existente 1`] = `
<div>
  <div>
    <div
      class="row rows-cols-1 row-cols-md-3 g-3"
    >
      <div
        class="col animate__fadeIn animate__animated"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-4"
            >
              <img
                alt="Spider Man"
                class="card-img"
                src="/assets/heroes/marvel-spider.jpg"
              />
            </div>
            <div
              class="col-8"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Spider Man
                </h5>
                <p
                  class="card-text"
                >
                  Peter Parker
                </p>
                <p
                  class="card-text text-muted"
                >
                  Amazing Fantasy #15
                </p>
                <a
                  href="/hero/marvel-spider"
                >
                  Más...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col animate__fadeIn animate__animated"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-4"
            >
              <img
                alt="Captain America"
                class="card-img"
                src="/assets/heroes/marvel-captain.jpg"
              />
            </div>
            <div
              class="col-8"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Captain America
                </h5>
                <p
                  class="card-text"
                >
                  Steve Rogers
                </p>
                <p
                  class="card-text text-muted"
                >
                  Captain America Comics #1
                </p>
                <a
                  href="/hero/marvel-captain"
                >
                  Más...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col animate__fadeIn animate__animated"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-4"
            >
              <img
                alt="Iron Man"
                class="card-img"
                src="/assets/heroes/marvel-iron.jpg"
              />
            </div>
            <div
              class="col-8"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Iron Man
                </h5>
                <p
                  class="card-text"
                >
                  Tony Stark
                </p>
                <p
                  class="card-text text-muted"
                >
                  Tales of Suspense #39
                </p>
                <a
                  href="/hero/marvel-iron"
                >
                  Más...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col animate__fadeIn animate__animated"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-4"
            >
              <img
                alt="Thor"
                class="card-img"
                src="/assets/heroes/marvel-thor.jpg"
              />
            </div>
            <div
              class="col-8"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Thor
                </h5>
                <p
                  class="card-text"
                >
                  Thor Odinson
                </p>
                <p
                  class="card-text text-muted"
                >
                  Journey into Myster #83
                </p>
                <a
                  href="/hero/marvel-thor"
                >
                  Más...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col animate__fadeIn animate__animated"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-4"
            >
              <img
                alt="Hulk"
                class="card-img"
                src="/assets/heroes/marvel-hulk.jpg"
              />
            </div>
            <div
              class="col-8"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Hulk
                </h5>
                <p
                  class="card-text"
                >
                  Bruce Banner
                </p>
                <p
                  class="card-text text-muted"
                >
                  The Incredible Hulk #1
                </p>
                <a
                  href="/hero/marvel-hulk"
                >
                  Más...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col animate__fadeIn animate__animated"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-4"
            >
              <img
                alt="Wolverine"
                class="card-img"
                src="/assets/heroes/marvel-wolverine.jpg"
              />
            </div>
            <div
              class="col-8"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Wolverine
                </h5>
                <p
                  class="card-text"
                >
                  James Howlett
                </p>
                <p
                  class="card-text text-muted"
                >
                  The Incredible Hulk #180
                </p>
                <a
                  href="/hero/marvel-wolverine"
                >
                  Más...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col animate__fadeIn animate__animated"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-4"
            >
              <img
                alt="Daredevil"
                class="card-img"
                src="/assets/heroes/marvel-daredevil.jpg"
              />
            </div>
            <div
              class="col-8"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Daredevil
                </h5>
                <p
                  class="card-text"
                >
                  Matthew Michael Murdock
                </p>
                <p
                  class="card-text text-muted"
                >
                  Daredevil #1
                </p>
                <a
                  href="/hero/marvel-daredevil"
                >
                  Más...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col animate__fadeIn animate__animated"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-4"
            >
              <img
                alt="Hawkeye"
                class="card-img"
                src="/assets/heroes/marvel-hawkeye.jpg"
              />
            </div>
            <div
              class="col-8"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Hawkeye
                </h5>
                <p
                  class="card-text"
                >
                  Clinton Francis Barton
                </p>
                <p
                  class="card-text text-muted"
                >
                  Tales of Suspense #57
                </p>
                <a
                  href="/hero/marvel-hawkeye"
                >
                  Más...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col animate__fadeIn animate__animated"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-4"
            >
              <img
                alt="Cyclops"
                class="card-img"
                src="/assets/heroes/marvel-cyclops.jpg"
              />
            </div>
            <div
              class="col-8"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Cyclops
                </h5>
                <p
                  class="card-text"
                >
                  Scott Summers
                </p>
                <p
                  class="card-text text-muted"
                >
                  X-Men #1
                </p>
                <a
                  href="/hero/marvel-cyclops"
                >
                  Más...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col animate__fadeIn animate__animated"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-4"
            >
              <img
                alt="Silver Surfer"
                class="card-img"
                src="/assets/heroes/marvel-silver.jpg"
              />
            </div>
            <div
              class="col-8"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Silver Surfer
                </h5>
                <p
                  class="card-text"
                >
                  Norrin Radd
                </p>
                <p
                  class="card-text text-muted"
                >
                  The Fantastic Four #48
                </p>
                <a
                  href="/hero/marvel-silver"
                >
                  Más...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
